src/jit/lower.cpp

```diff
@@ -462,6 +462,17 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             hiOp2->gtNext      = hiResult;
             hiResult->gtPrev   = hiOp2;
 
+            if (oper == GT_ADD || oper == GT_SUB)
+            {
+                if (loResult->gtOverflow())
+                {
+                    hiResult->gtFlags |= GTF_OVERFLOW;
```

Comment:
You should be able to find a preexisting test that has the GTF_OVERFLOW bit set of LONG, and look at the generated code. Maybe shouldn't be hard to generate a test case that does overflow-checking long that also generates overflow from the low 32 bits, e.g., adding 0x80000000 + 0x80000000.